pkg/dyninst/actuator/state.go

```diff
@@ -914,3 +920,72 @@ func handleShutdown(sm *state, effects effectHandler) error {
 	}
 	return nil
 }
+
+func handleHeartbeatCheck(sm *state, effects effectHandler) {
+	now := time.Now()
+	interval := now.Sub(sm.lastHeartbeat)
+	sm.lastHeartbeat = now
+
+	totalCostSPS := 0.0
+	maxCostSPS := -1.0
+	var maxProg *program
+	detachedAny := false
+	for _, prog := range sm.programs {
+		if prog.state != programStateLoaded {
+			continue
+		}
+		proc, ok := sm.processes[prog.processKey]
+		if !ok {
+			// Not attached.
+			continue
+		}
```

Comment:
I think we also want to check if the proc is in state `processStateAttached`. If we're attaching it'd be maybe weird.